### DISCUSSION TITLE
Don't cache function returning with error code?

### DISCUSSION THREAD

**Discussion by kumy** [dimo414/bash-cache#30] [Q&A] [answered]
Hi,

Is there a simple way to not cache the function results if the function returns with an exit code > 0?

Use case, a function doing a network operation and cache it for 24h. If unluckily there is a network outage the first time the function is called, then every subsequent calls will return the same error.

I've quickly tried with the new `bc::force::`, but the logic to retry multiple times may complicate usage.

Thanks

Small example:
```bash
#!/usr/bin/env bash
set -euo pipefail

source bash-cache.sh

function func1() {
  date
  ## Simulate a transient error like a temporary network failure
  ret=$((($RANDOM)%2))
  echo "return $ret"
  return $ret
} && bc::cache func1 24h 24h

func1 || echo "Failed 1"

# Try to autorecover
func1 || { echo "Failed 2 -> retry"; bc::force::func1; } || echo "Still failing"
func1 || echo "Failed 3"

```

**Comment by kumy** [comment, marked as answer]
:man_facepalming: Seems a bug is already open for that https://github.com/dimo414/bash-cache/issues/25

**Reply by dimo414** [reply]
No worries, thanks for joining the conversation!
